### PR TITLE
[mds-provider-processor] A few unit tests for mds-provider-processor.

### PR DESCRIPTION
**Pull request by janedotx**
## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: packages/mds-provider-processor/tests/metrics.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      await seedDeviceState(event as VEHICLE_EVENT)
+    })
+  )
+  await seedDeviceState(VEHICLE_EVENTS.register)
+  await seedDeviceState(VEHICLE_EVENTS.register)
+}
+
+describe('Proc Metrics', () => {
+  before(async () => {
+    await db.initialize()
```
Comment (drtyh2o) [inline packages/mds-provider-processor/tests/metrics.test.ts]: Is a `cache.initialize()` also required, I see there is a `cache.shutdown()` call below.

File: packages/mds-provider-processor/tests/metrics.test.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    await seedDeviceStates()
+  })
+
+  after(async () => {
+    await db.shutdown()
+    await cache.shutdown()
+  })
+
+  it('calcEventCounts', async () => {
```
Comment (drtyh2o) [inline packages/mds-provider-processor/tests/metrics.test.ts]: This `async` unit test pattern is so much cleaner. We should stop writing tests using a `done()` callback.

File: packages/mds-provider-processor/package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "start": "yarn watch server",
     "test": "yarn test:eslint && yarn test:unit",
     "test:eslint": "eslint --fix --ignore-path ../../.gitignore '**/*.ts'",
-    "test:unit": "DOTENV_CONFIG_PATH=../../.env nyc --lines 80 ts-mocha --project ../../tsconfig.json",
+    "test:unit": "DOTENV_CONFIG_PATH=../../.env nyc --lines 60 ts-mocha --project ../../tsconfig.json",
```
Comment (drtyh2o) [inline packages/mds-provider-processor/package.json]: Curious how we can add new tests, but decrease coverage?

Comment (janedotx) [reply packages/mds-provider-processor/package.json]: I'm really not sure how this works, considering there were no tests before. Seems like a bug in mocha.

File: packages/mds-provider-processor/tests/metrics.test.ts (hunk 1 above)
Comment (janedotx) [reply packages/mds-provider-processor/tests/metrics.test.ts]: No, any time you make a call to the cache it tries to initialize it.

**Pull request merged**